FloripaPublicTransportationTests/RestApiTests.swift

```diff
@@ -0,0 +1,343 @@
+            },
+            completionTests: { routes in
+                XCTAssertEqual(routes.count, 0, "The returned array must be empty")
+        })
+    }
+    
+    func testDictionaryFindRoutesByStopName() {
+        testValidFindRoutesByStopName({ _ in
+            let obj = ["rows":[["id":22,"shortName":"131","longName":"AGRONÔMICA VIA GAMA D'EÇA","lastModifiedDate":"2009-10-26T02:00:00+0000","agencyId":9]], "rowsAffected":"0"]
+            return OHHTTPStubsResponse(JSONObject: obj, statusCode:200, headers:nil)
+            },
+            completionTests: { routes in
+                XCTAssertEqual(routes.count, 1, "The returned array must have 1 route")
+                XCTAssertEqual(routes[0].id, 22, "The returned route must have id=22")
+        })
+    }
+    
+    func testRealJsonFindRoutesByStopName() {
+        testValidFindRoutesByStopName({ _ in
+            let stubData = NSData(contentsOfFile: NSBundle.mainBundle().pathForResource("findRoutes", ofType: "json")!)
+            return OHHTTPStubsResponse(data: stubData!, statusCode:200, headers:nil)
+            },
+            completionTests: { routes in
+                XCTAssertEqual(routes.count, 2, "The returned array must have 2 routes")
+                XCTAssertTrue(routes.contains({ $0.id == 22 }) && routes.contains({ $0.id == 32 }), "The returned array must contain routes with ids = 22 and 32")
+        })
+    }
+    
+    // MARK: Generic functions
+    
+    /// Generic function to be used by tests of `findRoutesByStopName` that use a valid JSON
+    /// - Parameter stubResponse: the block that returns an `OHHTTPStubsResponse` with stubbed NSData/NSDictionary
+    /// - Parameter completionTests: the block of tests to be executed against the returned array
+    private func testValidFindRoutesByStopName(stubResponse: OHHTTPStubsResponseBlock, completionTests: (routes: [Route]) -> Void)
+    {
+        testFindRoutesByStopName(stubResponse) { routes in
+            XCTAssertNotNil(routes, "The returned array must not be nil")
+            completionTests(routes: routes!)
+        }
+    }
+    
+    /// Generic function to be used by tests of `findRoutesByStopName` that use an invalid JSON
+    /// (or none at all)
+    /// - Parameter stubResponse: the block that returns an `OHHTTPStubsResponse` with stubbed NSData/NSDictionary
+    private func testInvalidFindRoutesByStopName(stubResponse: OHHTTPStubsResponseBlock)
+    {
+        let nsLogArrayCountBefore = SystemLogAccessor.NSLogArray().count
+        testFindRoutesByStopName(stubResponse) { routes in
+            XCTAssertNotNil(routes, "The returned array must not be nil")
+            XCTAssertEqual(routes!.count, 0, "The returned array must be empty")
+            XCTAssertGreaterThan(SystemLogAccessor.NSLogArray().count, nsLogArrayCountBefore, "This application should have increased its number of logs due to the detected errors")
```

Comment:
It was a quick way to assert that the application was logging when those errors occurred. It's not really for checking the tests, it's just an improvement over the test, like: nothing should crash AND something must be logged. Just to be sure that the app itself is aware of the failure.